src/pages/LoadingPage.tsx

```diff
@@ -1,3 +1,70 @@
-const LoadingPage: React.FC = () => <div>I am the loading page</div>;
+import { useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { LogTypes } from "constants/enums";
+import {
+  getEvergreenTaskLogURL,
+  getEvergreenTestLogURL,
+  getResmokeLogURL,
+} from "constants/logURLTemplates";
+import { slugs } from "constants/routes";
+import { useLogContext } from "context/LogContext";
+import { useAxiosGet } from "hooks/useAxios";
+
+interface LoadingPageProps {
+  onLoad: () => void;
+  logType: LogTypes;
+}
+
+const LoadingPage: React.FC<LoadingPageProps> = ({ onLoad, logType }) => {
+  const {
+    [slugs.buildID]: buildID,
+    [slugs.origin]: origin,
+    [slugs.testID]: testID,
+    [slugs.taskID]: taskID,
+    [slugs.execution]: execution,
+  } = useParams();
+
+  const { ingestLines } = useLogContext();
+
+  let url = "";
+  switch (logType) {
+    case LogTypes.RESMOKE_LOGS: {
+      if (buildID && testID) {
+        url = getResmokeLogURL(buildID, testID);
+      } else if (buildID) {
+        url = getResmokeLogURL(buildID);
+      }
+      break;
+    }
+    case LogTypes.EVERGREEN_TASK_LOGS: {
+      if (!taskID || !execution || !origin) {
+        break;
+      }
+      url = getEvergreenTaskLogURL(taskID, execution, origin as any);
+      break;
+    }
+    case LogTypes.EVERGREEN_TEST_LOGS: {
+      if (!taskID || !execution || !testID) {
+        break;
+      }
+      url = getEvergreenTestLogURL(taskID, execution, testID);
+      break;
+    }
+    default:
+      break;
+  }
+
+  const { data, error } = useAxiosGet(url);
+  useEffect(() => {
+    if (data) {
+      ingestLines(data.split("\n"), logType);
+      onLoad();
+    }
+  }, [data, ingestLines, onLoad, logType]);
+  if (error) {
+    console.error(error);
+  }
+  return <div>I am the loading page</div>;
```

Comment:
(disclaimer: i did not test this)

I wonder if this needs to be in a hook... 🤔 It would be awesome if we could just do 

```suggestion
  if (data) {
    ingestLines(data.split("\n"), logType);
    onLoad();
  }
  if (error) {
    console.error(error);
  }
  return <div>I am the loading page</div>;
```

Reply:
useEffect is meant to handle side effects like network requests and stuff. Without the hook i think error would trigger every render i've moved it into the hook.